### PR TITLE
Added a DSA course in hindi

### PR DESCRIPTION
**Pull request by shruti-sen2004**
## What does this PR do?
Add resource(s)

## For resources
Added a DSA course in hindi

### Why is this valuable (or not)?
Yes

### How do we know it's really free?
It is on youtube.

## Checklist:
- [ x] Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/main/docs/CONTRIBUTING.md).
- [x ] [Search](https://ebookfoundation.github.io/free-programming-books-search/) for duplicates.
- [x ] Include author(s) and platform where appropriate.
- [ x] Put lists in alphabetical order, correct spacing.
- [ x] Add needed indications (PDF, access notes, under construction).
- [x ] Used an informative name for this pull request.

## Follow-up

- Check the status of GitHub Actions and resolve any reported warnings!

### CODE REVIEW

**Comment by shruti-sen2004** [comment]
@eshellman Please review my changes and merge my pull request.
Thank you

**Review by Ajay-singh1 — COMMENTED** [review]
LGTM!

File: courses/free-courses-hi.md [hunk 1]
```diff
@@ -76,6 +76,7 @@
 * [Binary Trees - Level 1](https://www.youtube.com/playlist?list=PL-Jc9J83PIiHYxUk8dSu2_G7MR1PaGXN4) - Pepcoding
 * [Bit Manipulation - Level 2](https://youtube.com/playlist?list=PL-Jc9J83PIiFJRioti3ZV7QabwoJK6eKe) - Pepcoding
 * [Data Structures and Algorithms Course in Hindi](https://www.youtube.com/playlist?list=PLu0W_9lII9ahIappRPN0MCAgtOu3lQjQi) - CodeWithHarry
+* [Data Structures and Algorithms for GATE — Complete Playlist](https://www.youtube.com/playlist?list=PLC36xJgs4dxFCQVvjMrrjcY3XrcMm2GHy) - GATE CSE LECTURES BY AMIT KHURANA
```
Comment (eshellman) [inline courses/free-courses-hi.md:79]: no ALL CAPS

Comment (shruti-sen2004) [reply courses/free-courses-hi.md:79]: Okay

**Comment by shruti-sen2004** [comment]
@eshellman I have made the changes

**Review by Thenlie — APPROVED** [review]

**Pull request merged**